app/src/features/surveys/sampling-information/periods/table/SamplingPeriodTable.tsx

```diff
@@ -82,6 +111,20 @@ export const SamplingPeriodTable = (props: ISamplingPeriodTableProps) => {
       field: 'end_time',
       headerName: 'End time',
       flex: 1
+    },
+    {
+      field: 'duration',
+      headerName: 'Duration',
+      flex: 1,
+      renderCell: (params) => {
+        const duration = calculateDuration(
```

Comment:
Nice! This is just preference but I think you could avoid the separate function and just put the function's code in the renderCell function. We could also render the text in a more intuitive way using the humanize function

```
renderCell: (params) => {
        const startDateTime = params.row.start_time
          ? dayjs(`${params.row.start_date} ${params.row.start_time}`)
          : dayjs(params.row.start_date);
        const endDateTime = params.row.end_time
          ? dayjs(`${params.row.end_date} ${params.row.end_time}`)
          : dayjs(params.row.end_date);

        // Calculate the difference in milliseconds
        const diff = endDateTime.diff(startDateTime);

        return dayjs.duration(diff, 'millisecond').humanize();
      }
}
```

Humanize requires the relativeTIme plugin, too: 
```
import duration from 'dayjs/plugin/duration';
import relativeTime from 'dayjs/plugin/relativeTime';

dayjs.extend(duration);
dayjs.extend(relativeTime);
```

Reply:
Ah, humanize actually displays "a few seconds" for > 30 seconds and "a few minutes" for < 3 minutes, which is confusing. I pushed up something more precise.